pandas/core/series.py

```diff
@@ -2303,10 +2305,12 @@ def combine_first(self, other):
         new_index = self.index.union(other.index)
         this = self.reindex(new_index, copy=False)
         other = other.reindex(new_index, copy=False)
+        if is_datetimelike(this) and not is_datetimelike(other):
+            other = to_datetime(other)
+
         # TODO: do we need name?
```

Comment:
is this comment still relevant?

Reply:
Comment no longer relevant, I'll remove.

What do you think of datetime coercion of other? As I noted my (too long) comment above, using to_datetime coerces ints as well as strings. Old code used the Series constructor which only coerced strings but not ints. Alternatively, we could remove the datetimelike coercion entirely.